packages/logging/src/log.js

```diff
@@ -589,12 +589,27 @@ Log.prototype.write = function(entry, options, callback) {
     options = {};
   }
 
-  this.decorateEntries_(arrify(entry), function(err, decoratedEntries) {
-    // Ignore errors (the API will speak up if it has an issue).
+  if (!options.resource) {
```

Comment:
This does pretty much what `metadata_.assignDefaultResource` does. Can we use that instead:

```js
this.metadata_.assignDefaultResource(function(err, entry) {
  // Ignore errors (the API will speak up if it has an issue).
  
  var decoratedEntries;

  try {
    // ...
});
```